File: modules/video-postprocess/src/VideoPostprocessModule.ts
import { NativeModule, requireNativeModule } from "expo";
import { VideoPostprocessModuleEvents, PostprocessOptions } from "./VideoPostprocess.types";

declare class VideoPostprocessModule extends NativeModule<VideoPostprocessModuleEvents> {
  processClip(inputURL: string, outputURL: string, options: PostprocessOptions): Promise<string>;
  processClips(inputURLs: string[], outputDir: string, options: PostprocessOptions): Promise<string[]>;
}

export default requireNativeModule<VideoPostprocessModule>("VideoPostprocess");

File: modules/video-postprocess/android/src/main/java/expo/modules/videopostprocess/VideoPostprocessModule.kt
package expo.modules.videopostprocess

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class VideoPostprocessModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("VideoPostprocess")

    Events("onProgress")

    AsyncFunction("processClip") { inputURL: String, outputURL: String, options: Map<String, Any> ->
      throw UnsupportedOperationException("VideoPostprocess is not yet implemented on Android")
    }

    AsyncFunction("processClips") { inputURLs: List<String>, outputDir: String, options: Map<String, Any> ->
      throw UnsupportedOperationException("VideoPostprocess is not yet implemented on Android")
    }
  }
}
